cmd/api/src/api/v2/assetgrouptags.go

```diff
@@ -225,7 +225,11 @@ func (s *Resources) CreateAssetGroupTagSelector(response http.ResponseWriter, re
 		}
 
 		if selector, err := s.DB.CreateAssetGroupTagSelector(request.Context(), assetTagId, actor, createSelectorRequest.Name, description, false, true, autoCertify, createSelectorRequest.Seeds); err != nil {
-			api.HandleDatabaseError(request, response, err)
+			if errors.Is(err, database.ErrDuplicateAGTagSelectorName) {
+				api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusConflict, api.ErrorResponseAssetGroupTagSelectorDuplicateName, request), response)
+			} else {
+				api.HandleDatabaseError(request, response, err)
+			}
```

Comment:
not blocking suggestion: you can elevate this up so it's not nested using the same else if structure to better match the current golang spec
```suggestion
		if selector, err := s.DB.CreateAssetGroupTagSelector(request.Context(), assetTagId, actor, createSelectorRequest.Name, description, false, true, autoCertify, createSelectorRequest.Seeds); errors.Is(err, database.ErrDuplicateAGTagSelectorName) {
			api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusConflict, api.ErrorResponseAssetGroupTagSelectorDuplicateName, request), response)
		} else if err != nil {
			api.HandleDatabaseError(request, response, err)
		} 
```